yt/frontends/ramses/io_utils.pyx

```diff
@@ -63,21 +75,25 @@ def read_amr(FortranFile f, dict headers,
 
             if ng == 0:
                 continue
-            # Skip grid index, 'next' and 'prev' arrays (they are used
-            # to build the linked list in RAMSES)
-            f.skip(3)
-
-            # Allocate more memory if required
-            if ng > buffer_size:
-                pos = np.empty((ng, 3), dtype="d")
-                buffer_size = ng
-
-            pos[:ng, 0] = f.read_vector("d") - nx
-            pos[:ng, 1] = f.read_vector("d") - ny
-            pos[:ng, 2] = f.read_vector("d") - nz
+            # Skip grid index, 'next' and 'prev' arrays and possibly
+            # records from previous iterations
+            record_len = (ng * INT32_SIZE + INT64_SIZE)
+            f.seek(record_len * 3, 1)
+
+            f.read_vector_inplace("d", <void*> &pos_view[0, 0])
+            f.read_vector_inplace("d", <void*> &pos_view[0, 1])
+            f.read_vector_inplace("d", <void*> &pos_view[0, 2])
+
+            for i in range(ng):
+                pos_view[i, 0] -= nx
+            for i in range(ng):
+                pos_view[i, 1] -= ny
+            for i in range(ng):
+                pos_view[i, 2] -= nz
```

Comment:
Any reason not to vectorize this?

```suggestion
            pos_view[:ng, 0] -= nx
            pos_view[:ng, 1] -= ny
            pos_view[:ng, 2] -= nz
```

Reply:
I think Corentin's code may actually be faster: by using a view on the raw buffer, he allows the compiler to do all the vectorisation *without any overhead from Python's runtime checks (bound checking here, most likely)*. (I have not actually checked here, but that's a common optimisation technique and I have seen it in action)

Reply:
oh cool :) I figured there must be a reason, good to know!

Reply:
OH and I didn't realize I was in a .pyx file here haha.

Reply:
So I do agree with Clement here, but one reason that prevented me from using the vectorized loop is that I get an error with the vectorized view:
```
yt/frontends/ramses/io_utils.pyx:87:20: Inplace operators not supported on memoryview slices
```